### PR TITLE
[Feature] add a little bit comments; add support for environment variables

### PR DESCRIPTION
**Pull request by jingpengw**
auto load the environment variable if there is no secret json files found.

### CODE REVIEW

File: cloudvolume/secrets.py [hunk 1]
```diff
@@ -113,6 +113,14 @@ def aws_credentials(bucket = '', service = 'aws'):
       with open(aws_credentials_path, 'r') as f:
         aws_credentials = json.loads(f.read())
       break
+  
+  if not aws_credentials:
```
Comment (william-silversmith) [inline cloudvolume/secrets.py]: I don't think I understand the use case here. When would an environment variable be present but not the file? I do think we should support CloudVolume accepting credentials in the constructor including the s3 endpoint per #170 Does this introduce any security issues that were not present with the files?

I guess AWS thinks it's ok: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Comment (jingpengw) [reply cloudvolume/secrets.py]: environment variable could be transferred by `docker run -e`, and also defined in the kubernetes yml file. 
also boto3 and awscli will load environment variable too, so we only need to define environment variable and all the aws tools will work.

**Comment by william-silversmith** [comment]
I think since it's a new feature its less likely to break things. We should
probably study s3 and Google so see how they handle it. People will
probably form their expectations from how those tools behave.

**Comment by jingpengw** [comment]
at least, they will check the ~/.boto/credential file.

Jingpeng Wu
Postdoc in Princeton Neuroscience Institute, Princeton University, NJ
website: jingpengw.github.io

**Pull request merged**